retrieval/scrape.go

```diff
@@ -133,12 +135,13 @@ type scrapePool struct {
 	loops   map[uint64]loop
 
 	// Constructor for new scrape loops. This is settable for testing convenience.
-	newLoop func(context.Context, scraper, func() storage.Appender, func() storage.Appender, log.Logger) loop
-
-	logger       log.Logger
-	maxAheadTime time.Duration
+	newLoop func(*Target, scraper) loop
 }
 
+const maxAheadTime = 10 * time.Minute
```

Comment:
Since this is no longer a configurable, probably worth adding a comment here to explain its purpose and why it's a hard-coded value.